File: Example/javaapp/src/main/java/com/stripe/example/javaapp/network/BackendService.java
package com.stripe.example.javaapp.network;

import com.stripe.example.javaapp.model.ConnectionToken;

import org.jetbrains.annotations.NotNull;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

/**
 * The `BackendService` interface handles the two simple calls we need to make to our backend.
 */
public interface BackendService {

    /**
     * Get a connection token string from the backend
     */
    @POST("connection_token")
    Call<ConnectionToken> getConnectionToken();

    /**
     * Create a new reader location
     */
    @FormUrlEncoded
    @POST("create_location")
    Call<Void> createLocation(
            @Field("display_name") @NotNull String displayName,
            @Field("address[line1]") @NotNull String line1,
            @Field("address[line2]") String line2,
            @Field("address[city]") String city,
            @Field("address[postal_code]") String postalCode,
            @Field("address[state]") String state,
            @Field("address[country]") @NotNull String country
    );

    /**
     * Capture a specific payment intent on our backend
     */
    @FormUrlEncoded
    @POST("capture_payment_intent")
    Call<Void> capturePaymentIntent(@Field("payment_intent_id") @NotNull String id);

    /**
     * Cancel a specific payment intent on our backend
     */
    @FormUrlEncoded
    @POST("cancel_payment_intent")
    Call<Void> cancelPaymentIntent(@Field("payment_intent_id") @NotNull String id);
}

File: Example/kotlinapp/src/main/java/com/stripe/example/network/BackendService.kt
package com.stripe.example.network

import com.stripe.example.model.ConnectionToken
import retrofit2.Call
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

/**
 * The `BackendService` interface handles the two simple calls we need to make to our backend.
 */
interface BackendService {

    /**
     * Get a connection token string from the backend
     */
    @POST("connection_token")
    fun getConnectionToken(): Call<ConnectionToken>

    /**
     * Create a new reader location
     */
    @FormUrlEncoded
    @POST("create_location")
    fun createLocation(
        @Field("display_name") displayName: String,
        @Field("address[line1]") line1: String,
        @Field("address[line2]") line2: String?,
        @Field("address[city]") city: String?,
        @Field("address[postal_code]") postalCode: String?,
        @Field("address[state]") state: String?,
        @Field("address[country]") country: String
    ): Call<Void>

    /**
     * Capture a specific payment intent on our backend
     */
    @FormUrlEncoded
    @POST("capture_payment_intent")
    fun capturePaymentIntent(@Field("payment_intent_id") id: String): Call<Void>

    /**
     * Cancel a specific payment intent on our backend
     */
    @FormUrlEncoded
    @POST("cancel_payment_intent")
    fun cancelPaymentIntent(@Field("payment_intent_id") id: String): Call<Void>
}
